src/main/java/io/hhplus/tdd/point/facade/PointFacade.java

```diff
@@ -0,0 +1,100 @@
+package io.hhplus.tdd.point.facade;
+
+import io.hhplus.tdd.point.dto.PointHistory;
+import io.hhplus.tdd.point.dto.UserPoint;
+import io.hhplus.tdd.point.service.PointHistoryService;
+import io.hhplus.tdd.point.service.PointService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+@Service
+@RequiredArgsConstructor
+public class PointFacade {
+    private final PointService pointService;
+    private final PointHistoryService pointHistoryService;
+
+    // 사용자별 작업 큐 및 Lock 관리
+    private final ConcurrentHashMap<Long, Queue<Runnable>> userQueues = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<Long, ReentrantLock> userLocks = new ConcurrentHashMap<>();
+
+    // 사용자별 Lock 가져오기
+    private ReentrantLock getUserLock(long userId) {
+        return userLocks.computeIfAbsent(userId, id -> new ReentrantLock());
+    }
+
+    // 사용자별 Queue 가져오기
+    private Queue<Runnable> getUserQueue(long userId) {
+        return userQueues.computeIfAbsent(userId, id -> new LinkedList<>());
+    }
+
+    // 작업 큐에 작업 추가 및 처리 시작
+    private void submitTask(long userId, Runnable task) {
+        Queue<Runnable> queue = getUserQueue(userId);
+
+        synchronized (queue) {
+            queue.add(task); // 작업을 큐에 추가
+            if (queue.size() == 1) {
+                processQueue(userId); // 큐가 비어 있었다면 처리 시작
+            }
+        }
+    }
+
+    // 큐 처리
+    private void processQueue(long userId) {
+        Queue<Runnable> queue = getUserQueue(userId);
+        ReentrantLock lock = getUserLock(userId);
+
+        new Thread(() -> {
+            while (true) {
+                Runnable task;
+                synchronized (queue) {
+                    task = queue.poll(); // 큐에서 작업을 갖고오기
+                    if (task == null) {
+                        return; // 큐가 비어 있으면 종료
+                    }
+                }
+                lock.lock();
+                try {
+                    task.run(); // 작업 실행
+                } finally {
+                    lock.unlock();
+                }
+            }
```

Comment:
```suggestion
            while (true) {
                Runnable task;
                synchronized (queue) {
                    task = queue.poll(); // 큐에서 작업을 갖고오기
                    if (task == null) {
                        return; // 큐가 비어 있으면 종료
                    }
                    task.run(); // 작업 실행
                }
            }
```

이렇게 해도 `유저 요청` 에 대한 `순차적 실행` 은 문제 없을 것 같은데 작성하신 코드의 목적은 task.run() 의 시간이 오래 걸려 다른 스레드에서 queue 에 접근하지 못 할 것을 염두하고 세세하게 섹터를 나눠서 동기화 목적으로 작성 하신건가요?

저는 이런 생각을 하지 못했었는데 많은 생각이 드네요. 👍